tests/Database/ModelTest.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+use October\Rain\Database\Model;
+use Illuminate\Database\Schema\Blueprint;
+use October\Rain\Support\Facades\Schema;
+
+class ModelTest extends TestCase
+{
+    public function testAddCasts()
+    {
+        $model = new TestModel();
+
+        $this->assertEquals(['id' => 'int'], $model->getCasts());
+
+        $model->addCasts(['foo' => 'int']);
+
+        $this->assertEquals(['id' => 'int', 'foo' => 'int'], $model->getCasts());
+    }
+
+    public function testFalse()
+    {
+        $model = new TestModel();
+
+        // Test base guarded property
+        $this->assertTrue($model->isGuarded('data'));
+
+        // Test variations on casing
+        $this->assertTrue($model->isGuarded('DATA'));
+
```

Comment:
@LukeTowers additional test for the reverse (lower case key, upper-case guarded value)
```suggestion
        $this->assertTrue($module->isGuarded('name'));

```